WalletWasabi/Services/WalletService.cs

```diff
@@ -630,6 +635,50 @@ public async Task<Block> GetOrDownloadBlockAsync(uint256 hash, CancellationToken
 					cancel.ThrowIfCancellationRequested();
 					try
 					{
+						// Try to get block information from local running Core node first.
+						try
+						{
+							if (LocalBitcoinCoreNode == null || !LocalBitcoinCoreNode.IsConnected)
+							{
+								DisconnectDisposeNullLocalBitcoinCoreNode();
+								LocalBitcoinCoreNode = Node.ConnectToLocal(Network);
+							}
+
```

Comment:
You should try to `VersionHandshake` with timeout here. If the node has no slot for incoming transaction, it hangs on VersionHandshake. (I think it also hang if in IBD)

Reply:
You can check https://github.com/dgarage/NBXplorer/blob/4ca4c0e249964b6d472e47eed0ab7c461f29e85f/NBXplorer/BitcoinDWaiter.cs#L496

In this code, I tried to load the longest chain from the node. I handle several error that I regularly saw, and provide a good error message. (I use RPC though to make my error message better, you don't have to do this)

Reply:
@NicolasDorier I've been coding this but I found what seems to be a limitation in the current version of NBitcoin because IMO we should check whether the node can get full blocks or not (NBitcoin supports that) however I think Wasabi could, or at least in most of the cases, work well with pruned nodes too (288 blocks could be enough in many cases) but It seems there is no way to check for limited node capability (there is no way to get access to the `VersionPayload`) could you suggest me how to check if the node provides full or limited full block service?

Reply:
Forget it. I found it.